### PR TITLE
HPCC-31591 Force vcpkg build to use exact version of tools

### PR DESCRIPTION
**Pull request by GordonSmith**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: dockerfiles/vcpkg/ubuntu-22.10.dockerfile [hunk 1]
```diff
@@ -1,20 +1,6 @@
...
-    r-cran-rcpp \
-    r-cran-rinside \
-    r-cran-inline
-
-RUN git config --global --add safe.directory '*'
-
-WORKDIR /hpcc-dev
-
 ENTRYPOINT ["/bin/bash", "--login", "-c"]
+
```
Comment (Michael-Gardner) [inline dockerfiles/vcpkg/ubuntu-22.10.dockerfile:5]: I'm assuming we moved these additions to the dockerfile image over into the vcpkg repository dockerfile/ files?

File: cmake_modules/vcpkg.cmake [hunk 2]
```diff
@@ -4,13 +4,12 @@ if ("${VCPKG_DONE}" STREQUAL "")
 set(VCPKG_FILES_DIR "${CMAKE_BINARY_DIR}" CACHE STRING "Folder for vcpkg download, build and installed files")
 set(CMAKE_TOOLCHAIN_FILE ${HPCC_SOURCE_DIR}/vcpkg/scripts/buildsystems/vcpkg.cmake)
 set(VCPKG_ROOT ${HPCC_SOURCE_DIR}/vcpkg)
-set(VCPKG_OVERLAY_PORTS ${HPCC_SOURCE_DIR}/vcpkg_overlays;${HPCC_SOURCE_DIR}/vcpkg/overlays)
 set(VCPKG_INSTALLED_DIR "${VCPKG_FILES_DIR}/vcpkg_installed")
-set(VCPKG_INSTALL_OPTIONS "--downloads-root=${VCPKG_FILES_DIR}/vcpkg_downloads;--x-buildtrees-root=${VCPKG_FILES_DIR}/vcpkg_buildtrees;--x-packages-root=${VCPKG_FILES_DIR}/vcpkg_packages")
+set(VCPKG_INSTALL_OPTIONS "--x-abi-tools-use-exact-versions;--downloads-root=${VCPKG_FILES_DIR}/vcpkg_downloads;--x-buildtrees-root=${VCPKG_FILES_DIR}/vcpkg_buildtrees;--x-packages-root=${VCPKG_FILES_DIR}/vcpkg_packages")
```
Comment (Michael-Gardner) [inline cmake_modules/vcpkg.cmake:8]: Was the behavior before this checking the vcpkgTools.xml file and just using that as the minimum version?

**Review by Michael-Gardner — APPROVED** [review]

File: cmake_modules/vcpkg.cmake (hunk 2 above)
Comment (GordonSmith) [reply cmake_modules/vcpkg.cmake:8]: Correct it was defaulting to the system version if it was the same or newer.  FYI this is also now set for 9.4.x (9.6.x was always using it).

File: dockerfiles/vcpkg/ubuntu-22.10.dockerfile (hunk 1 above)
Comment (GordonSmith) [reply dockerfiles/vcpkg/ubuntu-22.10.dockerfile:5]: Yes - its only a convenience for testing, both 9.4.x and 9.6.x are setup this way already.

**Pull request merged**